samples/remote_attestation/common/attestation.cpp

```diff
@@ -2,163 +2,195 @@
         goto exit;
     }
 
-    // 1)  Validate the report's trustworthiness
-    // Verify the remote report to ensure its authenticity.
-    result =
-        oe_verify_report(remote_report, remote_report_size, &parsed_report);
-    if (result != OE_OK)
+    // Initialize the verifier.
+    verifier_result = oe_verifier_initialize();
+    if (verifier_result != OE_OK)
     {
-        TRACE_ENCLAVE("oe_verify_report failed (%s).\n", oe_result_str(result));
+        TRACE_ENCLAVE("oe_verifier_initialize failed.");
         goto exit;
     }
 
-    // 2) validate the enclave identity's signed_id is the hash of the public
-    // signing key that was used to sign an enclave. Check that the enclave was
-    // signed by an trusted entity.
-    if (memcmp(parsed_report.identity.signer_id, m_enclave_mrsigner, 32) != 0)
+    // Validate the evidence's trustworthiness
+    // Verify the remote evidence to ensure its authenticity.
+    result = oe_verify_evidence(
+        &sgx_remote_uuid,
+        remote_evidence,
+        remote_evidence_size,
+        NULL,
+        0,
+        NULL,
+        0,
+        &claims,
+        &claims_length);
+    if (result != OE_OK)
     {
-        TRACE_ENCLAVE("identity.signer_id checking failed.");
         TRACE_ENCLAVE(
-            "identity.signer_id %s", parsed_report.identity.signer_id);
+            "oe_verify_evidence failed (%s).\n", oe_result_str(result));
+        goto exit;
+    }
+
+    uint8_t* value;
 
-        for (int i = 0; i < 32; i++)
+    // 1) Validate the signer id.
+    value = (uint8_t*)_find_claim(claims, claims_length, OE_CLAIM_SIGNER_ID);
+
+    if (memcmp(value, m_enclave_mrsigner, 32) != 0)
+    {
+        TRACE_ENCLAVE("signer_id checking failed.");
+        TRACE_ENCLAVE("signer_id %s", value);
+
+        for (int j = 0; j < 32; j++)
         {
             TRACE_ENCLAVE(
                 "m_enclave_mrsigner[%d]=0x%0x\n",
-                i,
-                (uint8_t)m_enclave_mrsigner[i]);
+                j,
+                (uint8_t)m_enclave_mrsigner[j]);
         }
 
         TRACE_ENCLAVE("\n\n\n");
 
-        for (int i = 0; i < 32; i++)
+        for (int j = 0; j < 32; j++)
         {
-            TRACE_ENCLAVE(
-                "parsedReport.identity.signer_id)[%d]=0x%0x\n",
-                i,
-                (uint8_t)parsed_report.identity.signer_id[i]);
+            TRACE_ENCLAVE("signer_id)[%d]=0x%0x\n", j, (uint8_t)value[j]);
         }
         TRACE_ENCLAVE("m_enclave_mrsigner %s", m_enclave_mrsigner);
         goto exit;
     }
 
-    // Check the enclave's product id and security version
-    // See enc.conf for values specified when signing the enclave.
-    if (parsed_report.identity.product_id[0] != 1)
-    {
-        TRACE_ENCLAVE("identity.product_id checking failed.");
-        goto exit;
-    }
+    // 2) Check the enclave's product id.
+    value = (uint8_t*)_find_claim(claims, claims_length, OE_CLAIM_PRODUCT_ID);
```

Comment:
value can be null here. As commented above, I'd prefer _find_claim to return oe_claim_t*. Same below.

Reply:
Fixed, same as above.